sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ConditionalExpressionSuite.scala

```diff
@@ -248,4 +248,18 @@ class ConditionalExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
       .contains("CASE WHEN ... THEN struct<x:int> WHEN ... THEN struct<y:int> " +
         "ELSE struct<z:int> END"))
   }
+
+  test("SPARK-27917 test semantic equals of CaseWhen") {
+    val attrRef = AttributeReference("ACCESS_CHECK", StringType)()
+    val aliasAttrRef = attrRef.withName("access_check")
+    // Test for Equality
+    val caseWhenObj1 = CaseWhen(Seq((attrRef, Literal("A"))))
+    var caseWhenObj2 = CaseWhen(Seq((aliasAttrRef, Literal("A"))))
+    assert(caseWhenObj1.semanticEquals(caseWhenObj2))
+    assert(caseWhenObj2.semanticEquals(caseWhenObj1))
+    // Test for inEquality
+    caseWhenObj2 = CaseWhen(Seq((attrRef, Literal("a"))))
+    assert(!caseWhenObj1.semanticEquals(caseWhenObj2))
+    assert(!caseWhenObj2.semanticEquals(caseWhenObj1))
```

Comment:
Since this is a test case for `CaseWhen`, could you add a test coverage for `elseValue`?

Reply:
thanks for noticing this out, I had missed this. Have updated the test and source code accordingly